ui/component/chat/livestream-hyperchats.jsx

```diff
@@ -0,0 +1,190 @@
+  }, [exchangeRate]);
+
+  const stickerSuperChats = hyperChatsByAmount && hyperChatsByAmount.filter(({ comment }) => !!parseSticker(comment));
+
+  const showMore = superChatTopTen && hyperChatsByAmount && superChatTopTen.length < hyperChatsByAmount.length;
+  const elRef: ElementRef<any> = React.useRef();
+  const [showTooltip, setShowTooltip] = React.useState(true);
+
+  const HorizontalScroll = () => {
+    React.useEffect(() => {
+      const el = elRef.current;
+      if (el) {
+        const onWheel = (e) => {
+          if (
+            !showTooltip ||
+            e.deltaY === 0 ||
+            (el.scrollLeft === 0 && e.deltaY < 0) ||
+            el.scrollLeft === el.scrollWidth
+          ) {
+            return;
+          }
+
+          if (showTooltip) {
+            setShowTooltip(false);
+            e.preventDefault();
+            let scrollSpace =
+              el.scrollLeft + e.deltaY * 2.5 < el.scrollWidth ? el.scrollLeft + e.deltaY * 2.5 : el.scrollWidth;
+            el.scrollTo({
+              left: scrollSpace,
+              behavior: 'smooth',
+            });
+            setTimeout(() => {
+              setShowTooltip(true);
+            }, 1000);
+          }
+        };
+        el.addEventListener('wheel', onWheel);
+        return () => el.removeEventListener('wheel', onWheel);
+      }
+    }, []);
+    return elRef;
+  };
+
+  return !superChatTopTen ? null : (
+    <Slider isMobile={isMobile} hyperchatsHidden={hyperchatsHidden}>
+      <div
+        ref={HorizontalScroll()}
+        className={classnames('livestream-hyperchats__wrapper', {
+          'livestream-hyperchats__wrapper--mobile': isMobile,
+        })}
+      >
+        <div className="livestream-hyperchats">
+          {superChatTopTen.map((hyperChat: Comment) => {
+            const { comment, comment_id, channel_url, support_amount, is_fiat } = hyperChat;
+            const isSticker = stickerSuperChats && stickerSuperChats.includes(hyperChat);
+            const stickerImg = <OptimizedImage src={getStickerUrl(comment)} waitLoad loading="lazy" />;
+            const basedAmount = is_fiat && exchangeRate ? support_amount : support_amount * 10 * exchangeRate;
+
+            return showTooltip ? (
+              <Tooltip disabled title={isSticker ? stickerImg : comment} key={comment_id}>
+                <div
+                  className={classnames('livestream-hyperchat', {
+                    'livestream-hyperchat--mobile': isMobile,
+                    'hyperchat-preview-level1': basedAmount >= 5,
+                    'hyperchat-preview-level2': basedAmount >= 10,
+                    'hyperchat-preview-level3': basedAmount >= 50,
+                    'hyperchat-preview-level4': basedAmount >= 100,
+                    'hyperchat-preview-level5': basedAmount >= 500,
+                  })}
+                >
+                  <ChannelThumbnail uri={channel_url} xsmall showMemberBadge />
```

Comment:
Assuming I'm looking at the right place, I do see the `key`, but based on the error message, perhaps it needs to be in the `div` rather than `Tooltip`.  Not critical, can leave it last or postpone -- just some React efficiency thing.

<img width="447" alt="image" src="https://user-images.githubusercontent.com/64950861/180143375-b43066d7-274f-4890-82fd-1e7d261e958e.png">

Reply:
Legacy? It's fixed.